File: java-back-server/src/main/java/space/borisgk/taxi/api/kafka/MessageConsumer.java
package space.borisgk.taxi.api.kafka;

import com.fasterxml.jackson.databind.ObjectMapper;
import org.dozer.Mapper;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.kafka.annotation.KafkaListener;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.stereotype.Component;
import space.borisgk.taxi.api.exception.ServerException;
import space.borisgk.taxi.api.model.dto.MessageDto;
import space.borisgk.taxi.api.model.entity.Message;
import space.borisgk.taxi.api.model.entity.Trip;
import space.borisgk.taxi.api.model.entity.User;
import space.borisgk.taxi.api.service.MessageService;
import space.borisgk.taxi.api.service.TripService;

import java.util.List;

@Component
public class MessageConsumer {

    private Logger logger = LoggerFactory.getLogger(TripConsumer.class);

    @Autowired
    private ObjectMapper om;
    @Autowired
    private MessageService messageService;
    @Autowired
    private KafkaTemplate<String, String> kafkaTemplate;
    @Autowired
    private Mapper mapper;

    @KafkaListener(topics = "request.message.create", groupId = "server-java")
    public void messageCreate(String payload) throws ServerException {
        try {
            // TODO move to aspect
            logger.info("Receive payload:");
            logger.info(payload);

            MessageDto messageDto = om.readValue(payload, MessageDto.class);
            Message message = mapper.map(messageDto, Message.class);
            message.setUser(User.builder().id(messageDto.getUserId()).build());
            message.setTrip(Trip.builder().id(messageDto.getTripId()).build());
            message = messageService.saveMessage(message);
            messageDto = mapper.map(message, MessageDto.class);

            // TODO высылать только сокетам, которые слушают чат
            kafkaTemplate.send("response.message.create", om.writeValueAsString(messageDto));
        }
        catch (Exception e) {
            throw new ServerException(e);
        }
    }

    @KafkaListener(topics = "request.message.getByTripId", groupId = "server-java")
    public void messageGetByTripId(String payload) throws ServerException {
        try {
            // TODO move to aspect
            logger.info("Receive payload:");
            logger.info(payload);

            Integer tripId = Integer.parseInt(payload);
            List<Message> messages = messageService.getAllByTripId(tripId);
            List<MessageDto> messageDtos = mapper.map(messages, List.class);
            for (int i = 0; i < messageDtos.size(); i++) {
                messageDtos.get(i).setUserId(messages.get(i).getUser().getId());
                messageDtos.get(i).setTripId(messages.get(i).getTrip().getId());
            }

            // TODO высылать только сокетам, которые слушают чат
            kafkaTemplate.send("response.message.getByTripId", om.writeValueAsString(messageDtos));
        }
        catch (Exception e) {
            throw new ServerException(e);
        }
    }
}

File: kotlin-back-server/src/main/kotlin/space/borisgk/taxi/api/kafka/MessageConsumer.kt
package space.borisgk.taxi.api.kafka

import com.fasterxml.jackson.databind.ObjectMapper
import org.dozer.Mapper
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.kafka.annotation.KafkaListener
import org.springframework.kafka.core.KafkaTemplate
import org.springframework.stereotype.Component
import space.borisgk.taxi.api.exception.ServerException
import space.borisgk.taxi.api.model.dto.MessageDto
import space.borisgk.taxi.api.model.entity.Message
import space.borisgk.taxi.api.model.entity.Trip
import space.borisgk.taxi.api.model.entity.User
import space.borisgk.taxi.api.service.MessageService
import space.borisgk.taxi.api.service.TripService

@Component
class MessageConsumer {

    private val logger = LoggerFactory.getLogger(TripConsumer::class.java!!)

    @Autowired
    private lateinit var om: ObjectMapper
    @Autowired
    private lateinit var messageService: MessageService
    @Autowired
    private lateinit var kafkaTemplate: KafkaTemplate<String, String>
    @Autowired
    private lateinit var mapper: Mapper

    @KafkaListener(topics = ["request.message.create"])
    @Throws(ServerException::class)
    fun messageCreate(payload: String) {
        try {
            var messageDto = om.readValue(payload, MessageDto::class.java)
            var message = mapper.map(messageDto, Message::class.java)
            message.user = User(id = messageDto.userId)
            message.trip = Trip(id = messageDto.tripId)
            message = messageService.saveMessage(message)
            messageDto = mapper.map(message, MessageDto::class.java)

            // TODO высылать только сокетам, которые слушают чат
            kafkaTemplate.send("response.message.create", om.writeValueAsString(messageDto))
        } catch (e: Exception) {
            throw ServerException(e)
        }

    }

    @KafkaListener(topics = ["request.message.getByTripId"])
    @Throws(ServerException::class)
    fun messageGetByTripId(payload: String) {
        try {
            val tripId = Integer.parseInt(payload)
            val messages = messageService.getAllByTripId(tripId)
            val messageDtos = mapper.map(messages, List::class.java) as List<MessageDto>
            for (i in messageDtos.indices) {
                messageDtos[i].userId = messages.get(i).user!!.id
                messageDtos[i].tripId = messages.get(i).trip!!.id
            }

            // TODO высылать только сокетам, которые слушают чат
            kafkaTemplate.send("response.message.getByTripId", om.writeValueAsString(messageDtos))
        } catch (e: Exception) {
            throw ServerException(e)
        }

    }
}
